test/acceptance/features/steps/steps.py

```diff
@@ -484,18 +516,21 @@ def validate_persistent_sb(context, sb_name):
         assert False, "Service Binding got updated"
 
 
-@given(u'Secret "{secret_name}" does not contain "{key}"')
-def check_secret_key(context, secret_name, key):
+@then(u'Secret does not contain "{key}"')
+def check_secret_key(context, key):
     openshift = Openshift()
+    secret_name = polling2.poll(lambda: get_sbr_secret_name(context), step=100, timeout=1000, ignore_exceptions=(ValueError,), check_success=check_result)
     json_path = f'{{.data.{key}}}'
     polling2.poll(lambda: openshift.get_resource_info_by_jsonpath("secrets", secret_name, context.namespace.name,
                                                                   json_path) == "",
                   step=5, timeout=120, ignore_exceptions=(binascii.Error,))
 
 
-@step(u'Secret "{secret_name}" is empty')
-def validate_secret_empty(context, secret_name):
+@step(u'Secret is empty')
+def validate_secret_empty(context):
     openshift = Openshift()
+    if "sbr_name" in context:
```

Comment:
should not we throw an error if `sbr_name` does not exist in the context?

Reply:
Done

Reply:
we are still not returning error - we return "False"

how about adding here assert stating that `"sbr_name" must be in context?

Reply:
Yes. That's much better. I will add that.